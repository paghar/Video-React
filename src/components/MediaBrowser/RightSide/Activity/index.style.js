//@flow
import styled from 'styled-components';
import { Generals, Gradient } from 'theme/colors';
import Icon from 'components/shared/Icon';
export const StyledThumb = styled.div`
  border-radius: 10px;
  background-color: ${Generals.vi};
  width: 3px !important;
  opacity: 0.3;
`;
export const StyledTrack = styled.div`
  height: 100%;
  right: 0px;
  background-color: ${Generals.v};
  margin-top: 5px;
  border-radius: 10px;
  width: 2px !important;
  padding: 5px 0px;
`;
export const Shadow = styled.div`
  width: 255px;
  height: 236px;
  opacity: 0.85;
  background-image: ${Gradient};
  display: flex;
  position: absolute;
  bottom: 0;
  pointer-events: none;
  right: 0;
`;
export const StyledIcon = styled(Icon)`
  margin: 0 auto;
  display: flex;
  align-self: flex-end;
  margin-bottom: 18px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  user-select: none;
`;
