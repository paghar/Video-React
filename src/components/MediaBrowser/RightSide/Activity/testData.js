import image1 from 'assets/image/christopher-campbell-rDEOVtE7vOs-unsplash.png';
import image2 from 'assets/image/christopher-campbell-rDEOVtE7vOs-unsplash.png';
import image3 from 'assets/image/christopher-campbell-rDEOVtE7vOs-unsplash.png';
const testData = [
  {
    avatarSrc: '',
    shortName: 'SP',
    name: '1Sara Parker',
    createdAt: 1571295775000,
    text: 'Uploaded "aportman family"',
    avatarColor: 'gray',
  },
  {
    avatarSrc: image2,
    name: '2Amili Hilton',
    shortName: 'AH',
    createdAt: 1571209375000,
    text: 'Removed 3 files',
    avatarColor: '#fff',
  },
  {
    avatarSrc: image3,
    name: '3Andrew Sunny',
    shortName: 'AS',
    createdAt: 1571209375000,
    text: 'Shared "aportman family" with Peter Brown.',
    avatarColor: '#fff',
  },
  {
    avatarSrc: image1,
    name: '4Pablo Neroda',
    shortName: 'PN',
    createdAt: 1571027260000,
    text: 'Commented on "winter fall"',
    avatarColor: '#fff',
  },
  {
    avatarSrc: image1,
    name: '5Pablo Neroda',
    createdAt: 1571016460000,
    text: 'Commented on "winter fall"',
    shortName: 'PN',
    avatarColor: '#fff',
  },
  {
    avatarSrc: image1,
    name: '6Pablo Neroda',
    shortName: 'PN',
    createdAt: 1571009260000,
    text: 'Commented on "winter fall"',
    avatarColor: '#fff',
  },
  {
    avatarSrc: image1,
    name: '7Pablo Neroda',
    shortName: 'PN',
    createdAt: 1571002060000,
    text: 'Commented on "winter fall"',
    avatarColor: '#fff',
  },
  {
    avatarSrc: image1,
    name: '8Pablo Neroda',
    shortName: 'PN',
    createdAt: 1570963070000,
    text: 'Commented on "winter fall"',
    avatarColor: '#fff',
  },
  {
    avatarSrc: image1,
    name: '8Pablo Neroda',
    ShortName: 'PN',
    createdAt: 1570955870000,
    text: 'Commented on "winter fall"',
    color: '#fff',
  },
  {
    avatarSrc: image1,
    name: '9Pablo Neroda',
    ShortName: 'PN',
    createdAt: 1570836460000,
    text: 'Commented on "winter fall"',
    color: '#fff',
  },
];
export default testData;
