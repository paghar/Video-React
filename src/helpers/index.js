//@flow

// Converts hex color to rgba string
export function hexToRgb(hex: string, opacity: number): string {
  const bigInt = parseInt(
    typeof hex === 'string' ? hex.replace('#', '') : hex,
    16
  );
  const r = (bigInt >> 16) & 255;
  const g = (bigInt >> 8) & 255;
  const b = bigInt & 255;

  return `rgba(${[r, g, b].join(',')}, ${opacity})`;
}

// finds an element parent which has specific tag name
export const findParent = (
  element: HTMLElement | EventTarget,
  parentTag: string
): any => {
  let currentNode = element;
  while (currentNode !== null) {
    if (currentNode.tagName === parentTag) {
      return currentNode;
    }
    if (currentNode.parentNode) currentNode = currentNode.parentNode;
    else currentNode = null;
  }
  return currentNode;
};
