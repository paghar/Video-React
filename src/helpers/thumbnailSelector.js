import FixImage from 'components/MediaBrowser/AssetCard/CardThumbnails/FixImage';
import Unknown from 'components/MediaBrowser/AssetCard/CardThumbnails/Unknown';
import VideoSequences from 'components/MediaBrowser/AssetCard/CardThumbnails/VideoSequences';

export const FixImageType = 'FixImageType';
export const UnknownType = 'UnknownType';
export const VideoSequencesType = 'VideoSequenceType';

export const selectThumbnailComponent = thumbnailType => {
  switch (thumbnailType) {
    case FixImageType:
      return FixImage;
    case UnknownType:
      return Unknown;
    case VideoSequencesType:
      return VideoSequences;
    default:
      return Unknown;
  }
};
