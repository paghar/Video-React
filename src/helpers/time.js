//@flow
import type { ActivityType } from 'components/MediaBrowser/RightSide/Activity/ActivityItem';
//Get Day of (activity.createdAt)
export const getActivityTime = (activitiy: ActivityType): number => {
  return parseInt(new Date(activitiy.createdAt).getDay());
};
//Get Day of today
export const getToday = (): number => {
  return parseInt(new Date(Date.now()).getDay());
};
//minute of time gets to 2 digits
export const twoDigitMinutes = (time: number): string => {
  if (time > 0) {
    const minute = new Date(time).getMinutes();
    return `${minute < 10 ? '0' + minute : minute}`;
  }
  return '';
};
//hours of time gets to 2 digits
export const twoDigitHours = (time: number): string => {
  if (time > 0) {
    const Hours = new Date(time).getHours();
    return `${Hours < 10 ? '0' + Hours : Hours}`;
  }
  return '';
};
//turn time to Sat.oct9
export const getShortDate = (time: number): string => {
  const optionDay = { weekday: 'long' };
  const optionMonth = { month: 'long' };
  const day = new Intl.DateTimeFormat('en-US', optionDay)
    .format(time)
    .slice(0, 3);
  const month = new Intl.DateTimeFormat('en-US', optionMonth)
    .format(time)
    .slice(0, 3)
    .toLowerCase();
  const date = new Date(time).getDate();
  return `${day}.${month} ${date}`;
};
//turn time to 20th-FEB-22:22 or 2017,FEB 20
export const getListViewDate = (time: number): string => {
  const year = new Date(time).getFullYear();
  const thisYear = new Date(Date.now()).getFullYear();
  const date = new Date(time).getDate();
  const optionMonth = { month: 'long' };
  const month = new Intl.DateTimeFormat('en-US', optionMonth)
    .format(time)
    .slice(0, 3)
    .toUpperCase();

  const minutes = twoDigitMinutes(time);
  const hours = twoDigitHours(time);
  if (year === thisYear) return `${date}th-${month}-${hours}:${minutes}`;
  else return `${year},${month} ${date}`;
};
