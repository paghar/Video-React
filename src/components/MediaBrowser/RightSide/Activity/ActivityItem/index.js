//@flow
import React from 'react';
import {
  StyledDiv,
  UserNameLabel,
  TimeLabel,
  TextLabel,
  StyledActivityDiv,
} from './index.style';
import Avatar from 'components/shared/Avatar';
import { Display2, Display4, Display5 } from 'theme/fonts';
import { Generals } from 'theme/colors';
import { twoDigitMinutes } from 'helpers/time';

export type ActivityType = {
  avatarSrc: string,
  name: string,
  shortName: string,
  createdAt: number,
  text: string,
  avatarColor: string,
};
type Props = {
  className?: string,
};
const ActivityItem = ({
  avatarSrc,
  name,
  shortName,
  createdAt,
  text,
  avatarColor,
  className,
}: Props & ActivityType) => {
  return (
    <StyledDiv>
      <Avatar
        backgroundColor={avatarColor}
        imgUrl={avatarSrc}
        width="24px"
        height="24px"
        borderRadius="100%"
        font={Display4}
        color={Generals.viii}
      >
        {shortName}
      </Avatar>
      <StyledActivityDiv>
        <UserNameLabel font={Display2} color={Generals.vii}>
          {name}
        </UserNameLabel>
        <TextLabel font={Display4} color={Generals.vi}>
          {text}
        </TextLabel>
        <TimeLabel font={Display5} color={Generals.vi}>
          {`${twoDigitMinutes(createdAt) &&
            new Date(createdAt).getHours()}:${twoDigitMinutes(createdAt)}`}
        </TimeLabel>
      </StyledActivityDiv>
    </StyledDiv>
  );
};
export default ActivityItem;
