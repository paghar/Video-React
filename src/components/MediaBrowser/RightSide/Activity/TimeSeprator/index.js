//@flow
import React from 'react';
import { getShortDate } from 'helpers/time';
import { Time } from './index.style';
import { Display4 } from 'theme/fonts';
import { Generals } from 'theme/colors';
type Props = {
  time: number,
  className?: string,
};
const oneDay = 1000 * 60 * 60 * 24;
const TimeSeprator = ({ time, className }: Props) => {
  let today = new Date().setHours(0);
  today = new Date(today).setMinutes(0);
  today = new Date(today).setSeconds(0);
  const diff = parseInt(time - today);
  return (
    <Time className={className} font={Display4} hoverable color={Generals.vii}>
      {diff < oneDay && diff > 0
        ? 'Today'
        : diff < 0 && diff > -oneDay
        ? 'Yesterday'
        : getShortDate(time)}
    </Time>
  );
};

export default TimeSeprator;
