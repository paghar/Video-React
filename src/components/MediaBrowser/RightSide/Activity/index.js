//@flow
import React, { useState, useRef, useEffect } from 'react';
import ActivityItem from './ActivityItem';
import type { ActivityType } from './ActivityItem';
import TimeSeprator from './TimeSeprator';
import { getActivityTime, getToday } from 'helpers/time';
import { Scrollbars } from 'react-custom-scrollbars';
import { Generals } from 'theme/colors';
import { Shadow, StyledIcon, StyledThumb, StyledTrack } from './index.style';
type Props = {
  activities: Array<ActivityType>,
  className?: string,
};

const Activity = ({ activities, className }: Props) => {
  //Ref for ScrollBar
  const scrollBarRef = useRef();
  //State for showing shadow in scrollbar
  const [isShadowShown, setIsShadowShown] = useState(false);
  //detect if there is scrolling
  useEffect(() => {
    const currentScrollBar = scrollBarRef.current;
    if (currentScrollBar) {
      const scrollHeight = currentScrollBar.getScrollHeight();
      const clientHeight = currentScrollBar.getClientHeight();
      if (scrollHeight > clientHeight) {
        setIsShadowShown(true);
      }
    }
  }, [activities, setIsShadowShown]);
  //Hide the shadow box when scrolling
  const handleScrollStart = () => {
    setIsShadowShown(false);
  };
  //style of scroll thumb
  const renderThumb = ({ style }) => {
    return <StyledThumb style={style} />;
  };
  //style of scroll track
  const renderTrackVertical = ({ style }) => {
    return <StyledTrack style={style} />;
  };
  const activityList = activities.map((activity, id) => {
    const {
      name,
      createdAt,
      avatarSrc,
      text,
      avatarColor,
      shortName,
    } = activity;
    let prev = id ? getActivityTime(activities[id - 1]) : getToday();
    let current = getActivityTime(activities[id]);
    return (
      <React.Fragment key={id}>
        {(current !== prev || (id === 0 && current === prev)) && (
          <TimeSeprator time={createdAt} />
        )}
        <ActivityItem
          shortName={shortName}
          avatarSrc={avatarSrc}
          name={name}
          createdAt={createdAt}
          text={text}
          avatarColor={avatarColor}
        />
      </React.Fragment>
    );
  });
  return (
    <Scrollbars
      className={className}
      autoHide
      ref={ref => (scrollBarRef.current = ref)}
      renderThumbVertical={renderThumb}
      renderTrackVertical={renderTrackVertical}
      onScrollStart={handleScrollStart}
    >
      {activityList}
      {isShadowShown && (
        <Shadow>
          <StyledIcon fontSize="15px" color={Generals.vii} iconNumber="61560" />
        </Shadow>
      )}
    </Scrollbars>
  );
};

export default Activity;
