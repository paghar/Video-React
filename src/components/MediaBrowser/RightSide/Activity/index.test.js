//@flow
import React from 'react';
import Activity from '.';
import renderer from 'react-test-renderer';
import testData from './testData';
import { getToday, getActivityTime } from 'helpers/time';

describe('<Activity />', () => {
  it('checks the UI', () => {
    const wrapper = renderer
      .create(<Activity activities={testData} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('test the function', () => {
    expect(getActivityTime(testData[2])).toBe(3);
  });
  it('test the function', () => {
    expect(getActivityTime(testData[1])).toBe(3);
  });
});
