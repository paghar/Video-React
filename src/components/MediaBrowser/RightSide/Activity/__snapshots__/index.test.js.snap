// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Activity /> checks the UI 1`] = `
.c0 {
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c5 {
  pointer-events: auto;
}

.c2 {
  outline: none;
  border: none;
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 0px 27px 0px 20px;
  margin: 30px 0px;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-left: 7px;
  position: relative;
}

.c6 {
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  pointer-events: none;
}

.c8 {
  position: absolute;
  top: 0px;
  right: 0px;
}

.c7 {
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  pointer-events: none;
}

.c3 {
  background-color: gray;
  background-size: 100% 100%;
  width: 24px;
  height: 24px;
  border-radius: 100%;
  border: 0px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

.c9 {
  background-color: #fff;
  background-image: url(christopher-campbell-rDEOVtE7vOs-unsplash.png);
  background-size: 100% 100%;
  width: 24px;
  height: 24px;
  border-radius: 100%;
  border: 0px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

.c10 {
  background-color: background-image:url(christopher-campbell-rDEOVtE7vOs-unsplash.png);
  background-size: 100% 100%;
  width: 24px;
  height: 24px;
  border-radius: 100%;
  border: 0px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

.c1 {
  background: none;
  outline: none;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c12 {
  border-radius: 10px;
  width: 3px !important;
  opacity: 0.3;
}

.c11 {
  height: 100%;
  right: 0px;
  margin-top: 5px;
  border-radius: 10px;
  width: 2px !important;
  padding: 5px 0px;
}

<div
  style={
    Object {
      "height": "100%",
      "overflow": "hidden",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "bottom": 0,
        "left": 0,
        "marginBottom": 0,
        "marginRight": 0,
        "overflow": "scroll",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <span
      className="c0 c1"
    >
      Thu.oct 17
    </span>
    <div
      className="c2"
    >
      <button
        className="c3"
        height="24px"
        width="24px"
      >
        <span
          className="c0"
        >
          SP
        </span>
      </button>
      <div
        className="c4"
      >
        <span
          className="c5 c6"
        >
          1Sara Parker
        </span>
        <span
          className="c5 c7"
        >
          Uploaded "aportman family"
        </span>
        <span
          className="c5 c8"
        >
          10:32
        </span>
      </div>
    </div>
    <span
      className="c0 c1"
    >
      Wed.oct 16
    </span>
    <div
      className="c2"
    >
      <button
        className="c9"
        height="24px"
        width="24px"
      />
      <div
        className="c4"
      >
        <span
          className="c5 c6"
        >
          2Amili Hilton
        </span>
        <span
          className="c5 c7"
        >
          Removed 3 files
        </span>
        <span
          className="c5 c8"
        >
          10:32
        </span>
      </div>
    </div>
    <div
      className="c2"
    >
      <button
        className="c9"
        height="24px"
        width="24px"
      />
      <div
        className="c4"
      >
        <span
          className="c5 c6"
        >
          3Andrew Sunny
        </span>
        <span
          className="c5 c7"
        >
          Shared "aportman family" with Peter Brown.
        </span>
        <span
          className="c5 c8"
        >
          10:32
        </span>
      </div>
    </div>
    <span
      className="c0 c1"
    >
      Mon.oct 14
    </span>
    <div
      className="c2"
    >
      <button
        className="c9"
        height="24px"
        width="24px"
      />
      <div
        className="c4"
      >
        <span
          className="c5 c6"
        >
          4Pablo Neroda
        </span>
        <span
          className="c5 c7"
        >
          Commented on "winter fall"
        </span>
        <span
          className="c5 c8"
        >
          7:57
        </span>
      </div>
    </div>
    <div
      className="c2"
    >
      <button
        className="c9"
        height="24px"
        width="24px"
      />
      <div
        className="c4"
      >
        <span
          className="c5 c6"
        >
          5Pablo Neroda
        </span>
        <span
          className="c5 c7"
        >
          Commented on "winter fall"
        </span>
        <span
          className="c5 c8"
        >
          4:57
        </span>
      </div>
    </div>
    <div
      className="c2"
    >
      <button
        className="c9"
        height="24px"
        width="24px"
      />
      <div
        className="c4"
      >
        <span
          className="c5 c6"
        >
          6Pablo Neroda
        </span>
        <span
          className="c5 c7"
        >
          Commented on "winter fall"
        </span>
        <span
          className="c5 c8"
        >
          2:57
        </span>
      </div>
    </div>
    <div
      className="c2"
    >
      <button
        className="c9"
        height="24px"
        width="24px"
      />
      <div
        className="c4"
      >
        <span
          className="c5 c6"
        >
          7Pablo Neroda
        </span>
        <span
          className="c5 c7"
        >
          Commented on "winter fall"
        </span>
        <span
          className="c5 c8"
        >
          0:57
        </span>
      </div>
    </div>
    <span
      className="c0 c1"
    >
      Sun.oct 13
    </span>
    <div
      className="c2"
    >
      <button
        className="c9"
        height="24px"
        width="24px"
      />
      <div
        className="c4"
      >
        <span
          className="c5 c6"
        >
          8Pablo Neroda
        </span>
        <span
          className="c5 c7"
        >
          Commented on "winter fall"
        </span>
        <span
          className="c5 c8"
        >
          14:07
        </span>
      </div>
    </div>
    <div
      className="c2"
    >
      <button
        className="c10"
        height="24px"
        width="24px"
      />
      <div
        className="c4"
      >
        <span
          className="c5 c6"
        >
          8Pablo Neroda
        </span>
        <span
          className="c5 c7"
        >
          Commented on "winter fall"
        </span>
        <span
          className="c5 c8"
        >
          12:07
        </span>
      </div>
    </div>
    <span
      className="c0 c1"
    >
      Sat.oct 12
    </span>
    <div
      className="c2"
    >
      <button
        className="c10"
        height="24px"
        width="24px"
      />
      <div
        className="c4"
      >
        <span
          className="c5 c6"
        >
          9Pablo Neroda
        </span>
        <span
          className="c5 c7"
        >
          Commented on "winter fall"
        </span>
        <span
          className="c5 c8"
        >
          2:57
        </span>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "borderRadius": 3,
        "bottom": 2,
        "display": "none",
        "height": 6,
        "left": 2,
        "opacity": 0,
        "position": "absolute",
        "right": 2,
        "transition": "opacity 200ms",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,.2)",
          "borderRadius": "inherit",
          "cursor": "pointer",
          "display": "block",
          "height": "100%",
          "position": "relative",
        }
      }
    />
  </div>
  <div
    className="c11"
    style={
      Object {
        "display": "none",
        "opacity": 0,
        "position": "absolute",
        "transition": "opacity 200ms",
        "width": 6,
      }
    }
  >
    <div
      className="c12"
      style={
        Object {
          "display": "block",
          "position": "relative",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
